---
title: |
  Comparing antigenic distance metrics for influenza against genetic and
  antigenic history
subtitle: "EPID 8200 project proposal"
author:
  - W. Zane Billings
  - Amanda Skarlupka
  - Lambodhar Damodaran
  - Justin Bahl
  - Andreas Handel
date: last-modified
date-format: iso
format: docx
execute:
  echo: false
  message: false
  warning: false
---

```{r}
#| label: setup

library(gtsummary)
library(flextable)

# https://stackoverflow.com/questions/57175351/flextable-autofit-in-a-rmarkdown-to-word-doc-causes-table-to-go-outside-page-mar
FitFlextableToPage <- function(ft, pgwidth = 6){

  ft_out <- ft %>% autofit()

  ft_out <- width(ft_out, width = dim(ft_out)$widths*pgwidth /(flextable_dim(ft_out)$widths))
  return(ft_out)
}
```


# Project summary

Influenza A evolves rapidly and causes seasonal epidemics in temperate climates.
Subsequent epidemics are often induced by immune escape and the spread of a
novel variant. The antigenic distance between variants of influenza can be
calculated using multiple different methods, and is not always equivalent to
the genetic or evolutionary distance between to strains. There is no consensus
on which distance calculation methods most accurately map to immune
phenotypes and immune escape variants. We propose to calculate many different
distance metrics on the same database, including antigenic cartography. We can
then use the sequences to reconstruct a maximum likelihood evolutionary history,
and compare distance-based evolutionary trees within a likelihood framework to
determine which metrics most accurately reflect the evolutionary history of
the immune escape variant strains.

{{< pagebreak >}}

# Significance

The natural history of influenza is shaped predominantly by rapid evolution due
to host selective pressures, along with periodic recombination events which lead
to the proliferation of novel variants. We call these evolutionary processes
antigenic drift and antigenic shift respectively. Effectively vaccinating
against influenza requires careful consideration of these evolutionary processes,
and current vaccines rely on predicting variants of concern which are expected
to circulate in the upcoming season.

Current influenza vaccines rarely elicit strong responses to many different
strains of influenza, and so are less effective when predictions for circulating
strains are incorrect, or when novel viruses emerge due to recombination as in
2009. Developing a broadly protective influenza vaccine which responds to many
different strains of influenza is, therefore, a crucial goal for reduction of
the burden of current and future influenza epidemics.

However, measuring the breadth of a vaccine response is difficult, and many
proposed measures involve using the year of isolation of influenza viruses as
a proxy for sequence dissimilarity due to antigenic drift. Especially for H1N1,
which has two distinct historical lineages, year of isolation is an adequate
measure of antigenic distance. However, there are many different measures of
genetic and antigenic distance, with no clear consensus on which is best for
influenza. Comparing the similarity of antigenic distance measurements and their
ability to reconstruct the evolutionary history of influenza is necessary for
improving methods for quantifying the breadth of vaccine response.

# Approach

We will combine data from an influenza cohort study along with publicly avaiable
influenza sequence data in order to compare multiple measurements of genetic and
antigenic distance. The cohort data we will use is comprised of volunteers
enrolled at three different study sites from September 2013 ongoing. The study design is a prospective, open cohort study with one treatment
arm. Recruited individuals completed a questionnaire at enrollment to provide
demographic information (including sex, race, age, comorbidities, and prior
vaccination history), gave a pre-vaccination serum sample, and received a
seasonal influenza vaccine. Individuals returned for a followup visit with a
target timeframe of 21 or 28 days post-vaccination (depending on the year) and
gave a second serum sample. Individuals who were under the age of 65 received
the standard dose (SD) FluZone (Sanofi Pasteur) vaccine, while individuals who
were age 65 or older were allowed to choose either FluZone or FluZone High Dose
(HD). Individuals were allowed to return in subsequent years.

For our study, we selected records that had both a pre and post-vaccination
serum time point. We also only used records from study years 2013 up to March 2020, as the historical panel was reduced in subsequent years due to concerns arising from the COVID-19 pandemic. @Tbl-demographics shows demographics and counts for participants who were selected for inclusion in our study. We will
also only consider responses to influenza A, and we will treat H1N1 and H3N2
as independent lineages for the purpose of our analyses.

Researchers conducted hemagglutination inhibition (HAI) assays to a panel of
historical strains, which varied slightly over the duration of the study due to
funding and virus availibilities. The strains they used, along with the UniProt
and/or GISAID acession numbers for the protein sequences are given in
@tbl-sequence-numbers. The years for which each strain was used are shown in
@tbl-strain-years.

In order to compare genetic distance measures, we will first build a maximum
likelihood (ML) phylogenetic tree using the RNA sequences for each strain.
Then, we will calculate the pairwise distances between strains using several
different methods: Hamming distance, Levenshtein distance, Grantham's distance,
$p$-epitope distance, and normalized Euclidean distance from antigenic
cartography (see following section). Then, we can use each of the computed
distance matrices to build a distance-based tree via neighbor joining. The
likelihood of the resulting distance-based tree can be calculated under the
same framework as the ML tree, and we can use a likelihood ratio test to
determine whether the distance-based tree is significantly worse than the
ML tree. Note that we can easily extend our proposed methods to include any
number of different genetic or antigenic distance metrics, but as this is a
proof of concept study, we limited our concern to these metrics.

We will also check whether the ML tree reflects what we know about evolutionary
history. Since the strains we will use emerged in a specific and known order, we
can compare the inferred phylogeny based on sequence distance to the true
phylodynamic history. In particular, comparing the likelihoods of the ML and
cartography-based tree should give us insight into how similar the genetic and
antigenic patterns of evolution are for both H1N1 and H3N2 vaccine strains. We
can also compare the trees based on genetic distance measurements to the
tree based on the cartographic distance using a likelihood ratio test, so we can
determine which measures of genetic distance are similar to the genetic and
antigenic evolution patterns we detect in the data.

Finally, we will also conduct correlation and reliability analyses on the
different measures of antigenic distance to assess how similar they are to each
other. Our tree-based analysis can tell us which simple measures are best at
reconstructing the genetic or antigenic history of the virus. Susequent
correlation and reliability analyses can provide additional information about
how similar different measurements are and may help highlight why certain
measurements are more similar to the genetic or cartographic phylogenies.

We conducted antigenic cartography using the post-vaccination cohort data
described previously. We used the Racmacs R package to construct the
cartographic maps from the HAI data, which employs an iterative multidimensional
scaling (MDS) process to reduce the matrix of measurements into fewer dimensions,
from which we can easily calculate Euclidean distance measurements. We computed
cartographic maps in 1 -- 5 dimensions, and based on the relative error for each,
we determined that 2 dimensions were suitable, which matches previous findings.

{{< pagebreak >}}

# Data tables

```{r}
#| label: tbl-demographics
#| tbl-cap: |
#|   Age, sex, race, dose, and study site distributions for the selected study
#|   years. Notice that the study was conducted at the FL and PA study sites
#|   from fall 2013 through spring 2017, and at the UGA study site beginning in
#|   January 2017 onwards.
readr::read_rds(here::here("results", "sd-table1.Rds")) |>
	gtsummary::as_flex_table() |>
	flextable::compose(
		i = 1, j = 1, part = "header",
		value = flextable::as_paragraph("")
	) |>
	flextable::width(
		j = 1,
		width = .9
	) |>
	flextable::width(
		j = 2:8,
		width = 0.8
	)
```

{{< pagebreak >}}

```{r}
#| label: tbl-strain-years
#| tbl-cap: |
#|   Strains used for HAI assays on cohort study samples, and the seasons in
#|   which those strains were used.
short_names_used <-
	readr::read_rds(here::here("data", "viruses_used.rds")) |>
	dplyr::pull(strain_short)

years_tab <-
	readr::read_rds(here::here("results", "virus_accession.rds")) |>
	dplyr::filter(short_name %in% short_names_used) |>
	dplyr::select(short_name, c(`2014`:`2019`)) |>
	# Format the seasons nicely
	dplyr::rename_with(
		.fn = function(x) {
			n <- as.numeric(substring(x, 2, 4))
			o <- paste0(n, "/", n + 1)
			return(o)
			},
		.cols = -short_name
	)

col_matrix <- ifelse(
	apply(years_tab[, -1] == "x", c(1, 2), isTRUE),
	"gray",
	"white"
)

years_tab |>
	flextable::flextable() |>
	flextable::set_header_labels(
		values = list(
			short_name = "Strain"
		)
	) |>
	flextable::bg(
		j = -1,
		bg = col_matrix
	) |>
	flextable::color(
		j = -1,
		color = col_matrix
	) |>
	flextable::width(
		j = 1,
		width = .8
	) |>
	flextable::width(
		j = -1,
		width = .72
	) |>
	flextable::border_inner_v() |>
	flextable::border_inner_h()

```

{{< pagebreak >}}

```{r}
#| label: tbl-sequence-numbers
#| tbl-cap: |
#|   The strains used in the study along with the GISAID or UniProt accession
#|   numbers for the protein sequence of the virus stock used in the lab. For
#|   one sequence, the full length is not available. The nucleic acid sequences
#|   are also available for each of the strains, from the provided sources.
readr::read_rds(here::here("results", "virus_accession.rds")) |>
	dplyr::filter(short_name %in% short_names_used) |>
	dplyr::select(-c(`2014`:`2019`), -analysis_name) |>
	flextable::flextable() |>
	flextable::set_header_labels(
		values = list(
			full_strain_name = "Strain Name",
			#analysis_name = "Short Name",
			short_name = "Abbreviation",
			full_length = "Full Length?",
			ha_source = "HA Sequence Source"
		)
	) |>
	flextable::width(
		j = 1:4,
		width = c(2.7, 0.8, 1, 1.5)
	)
```

{{< pagebreak >}}

# References

<!-- END OF FILE -->
